Requires openssl/ssl.h despite SUPPORT_DTLS=OFF

while building vermont master@a1565d8 i got the following error:
```In file included from /tmp/vermont/src/modules/ipfix/IpfixExporterCfg.cpp:21:0:
/tmp/vermont/src/common/ipfixlolib/ipfixlolib_dtls.h:5:25: fatal error: openssl/ssl.h: Datei oder Verzeichnis nicht gefunden
 #include <openssl/ssl.h>
```


I used configure to set SUPPORT_DTLS=OFF and I do not have libssl-dev installed. I know this causes the error however I think openssl headers should only be required when configured this way.
Maybe related to #64, #83. Maybe this is already fixed in some of the new feature branches.

```
In file included from vermont/src/modules/ipfix/IpfixExporterCfg.cpp:21:0:
vermont/src/common/ipfixlolib/ipfixlolib_dtls.h:5:10: fatal error: openssl/ssl.h: No such file or directory
 #include <openssl/ssl.h>
          ^~~~~~~~~~~~~~~
compilation terminated.
src/modules/CMakeFiles/modules.dir/build.make:1502: recipe for target 'src/modules/CMakeFiles/modules.dir/ipfix/IpfixExporterCfg.cpp.o' failed
make[2]: *** [src/modules/CMakeFiles/modules.dir/ipfix/IpfixExporterCfg.cpp.o] Error 1
```

It looks like IpfixExporterCfg.cpp needs to conditionally on `#ifdef SUPPORT_DTLS` include ipfixlolib_dtls.h

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,16 +1,13 @@
 language: cpp
-compiler:
-  - clang
-  - gcc
 cache: ccache
 sudo: false
 
 addons:
   apt:
-    sources:
+    sources: &zeromq_source
     - sourceline: 'deb http://download.opensuse.org/repositories/network:/messaging:/zeromq:/release-stable/xUbuntu_14.04/ ./'
       key_url: 'http://download.opensuse.org/repositories/network:/messaging:/zeromq:/release-stable/xUbuntu_14.04/Release.key'
-    packages:
+    packages: &required_packages
       - cmake
       - libboost-dev
       - libboost-filesystem-dev
@@ -21,11 +18,45 @@ addons:
       - libpcap-dev
       - libsystemd-journal-dev
       - libsctp-dev
+    packages: &optional_packages
       - libssl-dev
       - libczmq-dev
 
-env:
-  - DTLS="ON" ZMQ="ON"
-  - DTLS="OFF" ZMQ="OFF"
+matrix:
+  include:
+    - compiler: gcc
+      sudo: true
+      before_install:
+        - sudo apt-get remove -qq libssl-dev
+      env: DTLS="OFF" ZMQ="OFF"
+      addons:
+        apt:
+          packages:
+            - *required_packages
+    - compiler: clang
+      sudo: true
+      before_install:
+        - sudo apt-get remove -qq libssl-dev
+      env: DTLS="OFF" ZMQ="OFF"
+      addons:
+        apt:
+          packages:
+            - *required_packages
+    - compiler: gcc
+      env: DTLS="ON" ZMQ="ON"
+      addons:
+        apt:
+          sources: *zeromq_source
+          packages:
+            - *required_packages
+            - *optional_packages
+    - compiler: clang
+      env: DTLS="ON" ZMQ="ON"
+      addons:
+        apt:
+          sources: *zeromq_source
+          packages:
+            - *required_packages
+            - *optional_packages
 
 script: cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DSUPPORT_JOURNALD=ON -DSUPPORT_DTLS="$DTLS" -DSUPPORT_ZMQ="$ZMQ" . && make && make test

--- a/src/modules/ipfix/IpfixExporterCfg.cpp
+++ b/src/modules/ipfix/IpfixExporterCfg.cpp
@@ -18,7 +18,10 @@
  *
  */
 
+#include "common/ipfixlolib/ipfixlolib_config.h"
+#ifdef SUPPORT_DTLS
 #include "common/ipfixlolib/ipfixlolib_dtls.h"
+#endif
 #include "IpfixExporterCfg.h"
 
 IpfixExporterCfg::IpfixExporterCfg(XMLElement* elem)
@@ -125,12 +128,15 @@ IpfixSender* IpfixExporterCfg::createInstance()
 				p->getPort());
 #endif
 		void *aux_config = NULL;
-		ipfix_aux_config_dtls_over_udp acdou;
-		ipfix_aux_config_dtls_over_sctp acdos;
 		ipfix_aux_config_udp acu;
 		ipfix_aux_config_udp *pacu = NULL;
+#ifdef SUPPORT_DTLS
+		ipfix_aux_config_dtls_over_udp acdou;
+		ipfix_aux_config_dtls_over_sctp acdos;
 		ipfix_aux_config_dtls *pacd = NULL;
+#endif
 		switch (p->getProtocol()) {
+#ifdef SUPPORT_DTLS
 			case DTLS_OVER_UDP:
 				acdou.max_connection_lifetime = dtlsMaxConnectionLifetime;
 				pacd = &acdou.dtls;
@@ -141,6 +147,7 @@ IpfixSender* IpfixExporterCfg::createInstance()
 				pacd = &acdos.dtls;
 				aux_config = &acdos;
 				break;
+#endif
 			case UDP:
 				aux_config = &acu;
 				pacu = &acu;
@@ -148,6 +155,7 @@ IpfixSender* IpfixExporterCfg::createInstance()
 			default:
 				break;
 		}
+#ifdef SUPPORT_DTLS
 		if (pacd) {
 			pacd->peer_fqdn = NULL;
 			const std::set<std::string> peerFqdns = p->getPeerFqdns();
@@ -155,6 +163,7 @@ IpfixSender* IpfixExporterCfg::createInstance()
 			if (it != peerFqdns.end())
 				pacd->peer_fqdn = it->c_str();
 		}
+#endif
 		if (pacu) {
 			pacu->mtu = p->getMtu();
 		}

